File: petcare_planner_backend/models/Task.js
const mongoose = require("mongoose");

const TaskSchema = new mongoose.Schema(
  {
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },
    taskTitle: {
      type: String,
      required: true,
    },
    petId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Pet",
      required: true,
    },
    taskType: {
      type: String,
      enum: ["Feeding", "Walking", "Grooming", "Medical"],
      required: true,
    },
    date: {
      type: Date,
      required: true,
    },
    time: {
      type: String,
      required: true,
    },
    repeat: {
      type: String,
      enum: ["None", "Daily", "Weekly", "Monthly"],
      default: "None",
    },
    notes: {
      type: String,
    },
    reminder: {
      type: Boolean,
      default: false,
    },
    status: {
      type: String,
      enum: ["pending", "completed"],
      default: "pending",
    },
  },
  {
    timestamps: true,
  }
);

module.exports = mongoose.model("Task", TaskSchema);

File: petcare_planner_frontend/lib/models/task.dart
import 'package:petcare_planner_frontend/models/pet.dart';

class Task {
  final String id;
  final String taskTitle;
  final Pet pet;
  final String taskType;
  final DateTime date;
  final String time;
  final String repeat;
  final String? notes;
  final bool reminder;

  Task({
    required this.id,
    required this.taskTitle,
    required this.pet,
    required this.taskType,
    required this.date,
    required this.time,
    this.repeat = "None",
    this.notes,
    this.reminder = false,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['_id'],
      taskTitle: json['taskTitle'],
      pet: Pet.fromJson(json['petId']),
      taskType: json['taskType'],
      date: DateTime.parse(json['date']),
      time: json['time'],
      repeat: json['repeat'] ?? "None",
      notes: json['notes'],
      reminder: json['reminder'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'taskTitle': taskTitle,
      'petId': pet,
      'taskType': taskType,
      'date': date.toIso8601String(),
      'time': time,
      'repeat': repeat,
      'notes': notes,
      'reminder': reminder,
    };
  }
}
